docs/book/src/clusterctl/commands/generate-yaml.md

```diff
@@ -0,0 +1,42 @@
+# clusterctl generate yaml
+
+The `clusterctl generate yaml` command processes yaml using clusterct's yaml
+processor.
+
+The intent of this command is to allow users who may have specific templates
+to leverage clusterctl's yaml processor for variable substitution. For
+example, this command can be leveraged in local and CI scripts or for
+development purposes.
```

Comment:
@CecileRobertMichon I added this section to the user documentation. Feel free to suggest edits 🙂